src/notifications/signal.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2021 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ *
+ */
+
+#ifndef WL_NOTIFICATIONS_SIGNAL_H
+#define WL_NOTIFICATIONS_SIGNAL_H
+
+#include <cassert>
+#include <functional>
+#include <list>
+#include <memory>
+#include <set>
+
+#include "base/macros.h"
+#include "base/wexception.h"
+
+namespace Notifications {
```

Comment:
I might remember incorrectly, but weren't namespaces lowercase in the Widelands code base?

Reply:
They're a mix, but mostly uppercase. Currently existing namespaces are
```
AddOns
AddOnsUI
AI
crypto
DebugConsole
FsMenu
Gl
i18n
JSON
LuaBases
LuaEditor
LuaGame
LuaGlobals
LuaMaps
LuaPath
LuaRoot
LuaTree
LuaUi
MapviewPixelFunctions
math
Notifications
QueueCmdFactory
RT
UI
Widelands
WLTestsuite
```
The `Notifications` namespace already exists, it houses `NotificationsManager` and `Subscriber` and I added `Signal` there because it fits semantically.